devito/propagator.py

```diff
@@ -213,6 +223,8 @@ def convert_equality_to_cgen(self, equality):
         else:
             s_lhs = ccode(self.time_substitutions(equality.lhs).xreplace(self._var_map))
             s_rhs = ccode(self.time_substitutions(equality.rhs).xreplace(self._var_map))
+            s_rhs = str(s_rhs).replace("pow", "powf")
+            s_rhs = str(s_rhs).replace("fabs", "fabsf")
```

Comment:
As mentioned above, does this need to be done after the `.xreplace(self._var_map)`? If not, we could apply this wholesale to all stencil and factor equalities in the constructor.